### PR TITLE
Mark `role` and `index` as required for chat schema

### PR DESCRIPTION
**Pull request by chenmoneygithub**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/chenmoneygithub/mlflow/pull/13279?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13279/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13279
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Fix chat schema. The current schema marks many fields as optional, which is incompatible with the schema enforced by agent frameworks.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/models/rag_signatures.py [hunk 1]
```diff
@@ -81,7 +88,7 @@ class StringResponse:
             type=Array(
                 Object(
                     [
-                        Property("role", DataType.string, required=False),
+                        Property("role", DataType.string),
```
Comment (dbczumar) [inline mlflow/models/rag_signatures.py:91]: @chenmoneygithub Are we sure that making these required is safe from a backwards compatibility perspective? cc @sunishsheth2009. Seems risky.

Comment (chenmoneygithub) [reply mlflow/models/rag_signatures.py:91]: Yea good call! i was consulting Sunish and we checked the code, `required` is the current behavior.  These schema files were added by me for clarity, but at my initial testing I wasn't aware of `databricks.agents.deploy` SDK, which caused an error.

Comment (sunishsheth2009) [reply mlflow/models/rag_signatures.py:91]: cc @prithvikannan who has more context on this and @bbqiu who is working on flexible schemas

Comment (prithvikannan) [reply mlflow/models/rag_signatures.py:91]: there is a small corner case here with mismatched mlflow versions. in `agents.deploy()` we compare the model's signature to the RAG signatures from MLflow to see if the model is considered "rag compatible". 

usually the MLflow version that is used when logging the model will be the same version installed when running `agents.deploy()`, but it is possible that the user logged with 2.16.2 and then later tries to deploy when 2.17.1 is installed. imo, this is a really small corner case but i think worth calling out

Comment (chenmoneygithub) [reply mlflow/models/rag_signatures.py:91]: Thanks for the inputs! With that context, I think we should proceed with this change?

**Comment by harupy** [comment]
@chenmoneygithub can you update the PR title and description? Why did we need this change?

**Review by harupy — APPROVED** [review]
LGTM

**Review by bbqiu — APPROVED** [review]
1. Could you please change the PR description to describe the potential incompatibility?
2. (nit) We use these schemas in the agent framework signature validation, so the current description is not entirely accurate. However, these fields do end up being required if they want to use any of our client UIs to chat with the deployed model, so this PR helps!

code LGTM -- will keep this top of mind if customer issues come up

**Pull request merged**